editor/components/block-toolbar/index.js

```diff
@@ -8,29 +8,40 @@ import { withSelect } from '@wordpress/data';
  */
 import './style.scss';
 import BlockSwitcher from '../block-switcher';
+import MultiBlocksSwitcher from '../block-switcher/multi-blocks-switcher';
 import BlockControls from '../block-controls';
 import BlockFormatControls from '../block-format-controls';
 
-function BlockToolbar( { block, mode } ) {
-	if ( ! block || ! block.isValid || mode !== 'visual' ) {
+function BlockToolbar( { blockUIDs, isValid, mode } ) {
+	if ( blockUIDs.length > 1 ) {
+		return (
+			<div className="editor-block-toolbar">
+				<MultiBlocksSwitcher />
+			</div>
+		);
+	}
+
+	if ( ! isValid || 'visual' !== mode ) {
 		return null;
 	}
 
 	return (
 		<div className="editor-block-toolbar">
-			<BlockSwitcher uids={ [ block.uid ] } />
+			<BlockSwitcher uids={ blockUIDs } />
 			<BlockControls.Slot />
```

Comment:
I wonder if it makes sense to keep the slots in case we want to show multi-selection controls or something like that in the future.